azerothcore/azerothcore-wotlk
# fix(Core/Spells): Resolve invalid spell casts on dead players when the cast is valid.

## Changes Proposed:

This PR proposes changes to:

There were two bugs present.
1. Ghosts (ex: Deathly Usher) were unable to detect other ghosts, fixed [here](https://github.com/SubstituteR/azerothcore-wotlk/blob/issues-20509/src/server/game/Entities/Creature/Creature.cpp#L1206).
2. Spells that did not require a dead player (which should allow casts on dead players) were not properly being checked, fixed [here](https://github.com/SubstituteR/azerothcore-wotlk/blob/21e7bcb46534a144eb960501d88787ad4382ffde/src/server/game/Spells/SpellInfo.cpp#L1778-L1782).

These changes were tested against Deathly Usher.

## Issues Addressed:

- 
Closes https://github.com/azerothcore/azerothcore-wotlk/issues/20509
Closes https://github.com/chromiecraft/chromiecraft/issues/2401
Closes https://github.com/azerothcore/azerothcore-wotlk/issues/9356

## SOURCE:

The changes have been validated through:

## Tests Performed:

This PR has been:


## How to Test the Changes:



1. Need to verify that enabling two-way visibility between ghosts ([line](https://github.com/SubstituteR/azerothcore-wotlk/blob/issues-20509/src/server/game/Entities/Creature/Creature.cpp#L1206)) does not create problems with other npcs / scripts / etc.

## Known Issues and TODO List:



## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
